backend/logic/deploy/records.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * Funcitonality for deployment records in the database
+ */
+const { ShopDeploymentStatuses } = require('../../enums')
+const { ShopDeployment, ShopDeploymentName } = require('../../models')
+const { assert } = require('../../utils/validators')
+const { getLogger } = require('../../utils/logger')
+
+const { DuplicateDeploymentError } = require('./common')
+
+const log = getLogger('logic.deploy.records')
+
+// Max age of a deployment before it is considered as failed.
+const MAX_PENDING_DEPLOYMENT_AGE = 10 * 60 * 1000 // 10 min
+
+/**
+ * Set a deployment as failed with given errorMessage
+ *
+ * @param deployment {object} - ShopDeployment model instance
+ * @param errorMessage {string} - Error message to store
+ */
+async function failDeployment(deployment, errorMessage) {
+  await deployment.update({
+    status: ShopDeploymentStatuses.Failure,
+    error: errorMessage
+  })
+}
+
+/**
+ * Set a deployment as succeeded
+ *
+ * @param deployment {object} - ShopDeployment model instance
+ * @param props {object} - Any extra model propreties to set
+ */
+async function passDeployment(deployment, props) {
+  await deployment.update({
+    status: ShopDeploymentStatuses.Success,
+    ...props
+  })
+}
+
+/**
+ * Create a shop deployment record
+ *
+ * @param shopId {number} - Shop ID to create a deployment for
+ * @param uuid {string} - A UUID identifier that can be used for later retrieval
+ * @param status {ShopDeploymentStatuses} - ShopDeploymentStatuses enum value
+ */
+async function createDeployment(
+  shopId,
+  uuid,
+  status = ShopDeploymentStatuses.Pending
```

Comment:
why does the status need to be passed? shouldn't a new deployment always get created as Pending? Like we would create the deployment row, then pass it's id to the deploymentProcessor.

Reply:
hypothetical future thinking, but really probably unnecessary.  It's doing no harm right now, but you're probably right.